unity-client/Assets/Scripts/MainScripts/DCL/Controllers/BuilderMode/BuilderEntity/DecentralandEntityToEdit.cs

```diff
@@ -0,0 +1,295 @@
+                renderer.material = editMaterial;
+                cont++;
+            }
+        }
+    }
+
+    void OnShapeUpdate(DecentralandEntity decentralandEntity)
+    {
+
+        if (IsSelected)
+            SaveOriginalMaterialAndSetEditMaterials();
+        CreateCollidersForEntity(decentralandEntity);
+        CheckIfEntityIsFloor();
+        CheckIfEntityIsVoxel();
+    }
+
+    void CreateCollidersForEntity(DecentralandEntity entity)
+    {
+        DecentralandEntity.MeshesInfo meshInfo = entity.meshesInfo;
+        if (meshInfo == null || meshInfo.currentShape == null)
+            return;
+        if (!meshInfo.currentShape.IsVisible())
+            return;
+        if (!meshInfo.currentShape.IsVisible() && meshInfo.currentShape.HasCollisions())
+            return;
+        if (!meshInfo.currentShape.IsVisible() && !meshInfo.currentShape.HasCollisions())
+            return;
+
+        if (!collidersDictionary.ContainsKey(entity.scene.sceneData.id + entity.entityId))
+        {
+            if (entity.children.Count > 0)
+            {
+                using (var iterator = entity.children.GetEnumerator())
+                {
+                    while (iterator.MoveNext())
+                    {
+                        CreateCollidersForEntity(iterator.Current.Value);
+                    }
+                }
+            }
+
+
+            GameObject entityCollider = new GameObject(entity.entityId);
+            entityCollider.layer = LayerMask.NameToLayer("OnBuilderPointerClick");
+
+
+            for (int i = 0; i < meshInfo.renderers.Length; i++)
+            {
+                Transform t = entityCollider.transform;
+                t.SetParent(meshInfo.renderers[i].transform);
+                t.ResetLocalTRS();
+
+                var meshCollider = entityCollider.AddComponent<MeshCollider>();
+
+                if (meshInfo.renderers[i] is SkinnedMeshRenderer)
+                {
+                    Mesh meshColliderForSkinnedMesh = new Mesh();
+                    (meshInfo.renderers[i] as SkinnedMeshRenderer).BakeMesh(meshColliderForSkinnedMesh);
+                    meshCollider.sharedMesh = meshColliderForSkinnedMesh;
+                    t.localScale = new Vector3(1 / entity.gameObject.transform.lossyScale.x, 1 / entity.gameObject.transform.lossyScale.y, 1 / entity.gameObject.transform.lossyScale.z);
+                }
+                else
+                {
+                    meshCollider.sharedMesh = meshInfo.renderers[i].GetComponent<MeshFilter>().sharedMesh;
+                }
+                meshCollider.enabled = meshInfo.renderers[i].enabled;
+
+            }
+
+            collidersDictionary.Add(entity.scene.sceneData.id + entity.entityId, entityCollider);
+        }
+    }
+
+    void SetLockStatus(bool isLocked)
+    {
+        this.isLocked = isLocked;
+        onStatusUpdate?.Invoke(this);
+    }
+
+    void CheckIfEntityIsFloor()
```

Comment:
The problem is that the entity can be unlocked even if it is a floor, this function only sets the initial state of the `IsLocked ` value based on if it is a floor. I agree with the name though, maybe `TestIfEntityIsFloor` is more descriptive?

Reply:
I would avoid `Test` in the name since it can be misleading but unfortunately I dont have a better alternative.

What I was suggesting was: 

```
public void Init(DecentralandEntity _entity, Material _editMaterial)
    {
        .
        .
        .

        if (rootEntity.meshRootGameObject && rootEntity.meshesInfo.renderers.Length > 0)
        {
            CreateCollidersForEntity(rootEntity);
            if(CheckIfEntityIsFloor())
                  SetLockStatus(true);  // <- This goes here
            CheckIfEntityIsVoxel();
        }
    }

    void bool CheckIfEntityIsFloor()
    {
        if (rootEntity.meshesInfo == null || rootEntity.meshesInfo.currentShape == null)
            return false;
        if (rootEntity.meshesInfo.renderers == null && rootEntity.meshesInfo.renderers.Length <= 0)
            return false;
        if (rootEntity.meshesInfo.mergedBounds.size.y >= 0.05f)
            return false;
        if (rootEntity.gameObject.transform.position.y >= 0.05f)
            return false;
        return true; 
    }
```